src/Microsoft.AspNet.JsonPatch/Adapters/ObjectAdapter.cs

```diff
@@ -734,100 +720,183 @@ public void Replace([NotNull] Operation operation, [NotNull] object objectToAppl
         /// <param name="objectToApplyTo">Object to apply the operation to.</param>
         public void Copy([NotNull] Operation operation, [NotNull] object objectToApplyTo)
         {
-            // get value at from location
-            object valueAtFromLocation = null;
-            var positionAsInteger = -1;
-            var actualFromProperty = operation.from;
+            // get value at from location and add that value to the path location
+            var valueAtFromLocationResult = GetValueAtLocation(operation.from, objectToApplyTo, operation);
 
-            positionAsInteger = GetNumericEnd(operation.from);
-
-            if (positionAsInteger > -1)
+            if (valueAtFromLocationResult.HasError)
             {
-                actualFromProperty = operation.from.Substring(0,
-                    operation.from.IndexOf('/' + positionAsInteger.ToString()));
+                // Return, error has already been logged in GetValueAtLocation
+                return;
             }
 
-            var patchProperty = FindPropertyAndParent(objectToApplyTo, actualFromProperty);
+            Add(operation.path,
+                valueAtFromLocationResult.PropertyValue,
+                objectToApplyTo,
+                operation);
+        }
+
+        /// <summary>
+        /// Method is used by Copy and Move to avoid duplicate code
+        /// </summary>
+        /// <param name="location">Location where value should be</param>
+        /// <param name="objectToGetValueFrom">Object to inspect for the desired value</param>
+        /// <param name="operationToReport">Operation to report in case of an error</param>
+        /// <returns>GetValueResult containing value and a bool signifying a possible error</returns>
```

Comment:
BTW there's no need for docs on private methods, or anything a user likely wouldn't ever call.